src/xtgeo/xyz/polygons.py

```diff
@@ -261,6 +237,32 @@ def dataframe(self):
     @dataframe.setter
     def dataframe(self, df):
         self._df = df.copy()
+        self._check_column_name()
+
+    def _check_column_name(self):
```

Comment:
Should probably rename this, as it doesn't only check, it modifies..

Reply:
Yes, feel free to change it 👍